src/ItemGrid.cpp

```diff
@@ -237,9 +240,15 @@ bool cItemGrid::IsSlotEmpty(int a_X, int a_Y) const
 
 void cItemGrid::Clear(void)
 {
+	if (IsCompletelyEmpty())
+	{
+		return;  // Already clear
+	}
+
+	auto Slots = GetSlots();
 	for (int i = 0; i < m_NumSlots; i++)
 	{
-		m_Slots[i].Empty();
+		Slots[i].Empty();
```

Comment:
Why not just free the allocated space of `unique_ptr`? This would also remove the if-condition.

Reply:
My thought on this was, that if the storage should become empty, then we want to deallocate the space again. E.g. If I copy a bunch of empty chests over a bunch of full chests, their memory should be freed.

Reply:
It's a balance between using more memory and wasting time allocating and deallocating repeatedly.
For example you might have a chest that clears and then fills with new loot when opened.  In which case the deallocation was just a waste of time.